src/nvim/os/stdpaths.c

```diff
@@ -69,13 +69,19 @@ const char *get_appname(void)
   return env_val;
 }
 
-/// Ensure that APPNAME is valid. In particular, it cannot contain directory separators.
+// Ensure that APPNAME is valid. In particular, it cannot contain .. surrounded
+// by path separators, the start of the string, or the end of the string.
 bool appname_is_valid(void)
 {
   const char *appname = get_appname();
   const size_t appname_len = strlen(appname);
+  if (appname[0] == PATHSEP) {
+    return false;
```

Comment:
`path_is_absolute`

Reply:
I'm not sure what you're meaning here, do you want me to set that variable instead of the early return?

Reply:
i'm saying `path_is_absolute` should be used instead of checking `appname[0] == PATHSEP`

Reply:
I did not know that was an option, that would have saved such a headache yesterday.
Thanks for the info :smile: updated it